tensorflow/core/kernels/softmax_op_gpu.cu.cc

```diff
@@ -72,28 +72,94 @@ struct softmax_traits<Eigen::half> {
   using accumulator_type = float;
 };
 
-template <typename T, typename U>
+template <typename T, typename U, int kUnroll>
```

Comment:
Nit: Do you expect to need other unroll values than 4 and 8? If no, we could replace the kUnroll parameter with a boolean "aligned" and just specialize on T = Eigen::half and aligned=true. It would save having two separate calls in the launcher below. I don't insist.